### DISCUSSION TITLE
How to add custom service worker when i use strategies 'generateSW'

### DISCUSSION THREAD

**Discussion by orangsed** [vite-pwa/vite-plugin-pwa#756] [Q&A]
I'm new to PWA technology, what I know is that there are 2 strategies in PWA, namely generateSW and injectManifest. 
From what I read, if you want to create a custom service worker, you should use injectManifest. what about using generateSW? 

I tried to use the importScripts provided by Vite PWA in the config file, but in the official documentation the use of importScript is not recommended. Then how do I create a custom service worker in the generateSW strategy without using the importScripts feature in the config file?

until now I still use importScript to add custom service workers which must be placed into the public folder

![image](https://github.com/user-attachments/assets/4bff22b7-3cf0-427f-af69-6d09202cfca8)
![image](https://github.com/user-attachments/assets/3a22d456-2176-4213-9386-6cff8f880ade)

**Comment by userquin** [comment]
You cannot use custom sw when using generateSW strategy. That strategy will generate the sw for you.

**Reply by orangsed** [reply]
how about use importScript? when i use that, it's can import my custom service worker in public folder

![image](https://github.com/user-attachments/assets/186cf15b-0c6c-4e75-80fd-e34165de9d33)
in the description above, I mean that I can add a custom service worker code such as push notification into sw which is automatically generated by the generateSW strategy.

in this case, I want to keep using the generateSW strategy by adding a sw for push notifications.

**Comment by userquin** [comment]
You can check the cjs scripts using workbox-build node module: the pwa plugin will do that work for you (check package.json scripts)

https://github.com/vite-pwa/workbox-window-es/tree/main/examples/vue-router/sw

**Reply by orangsed** [reply]
okay, thanks. i'll learn it

**Reply by userquin** [reply]
If you are building the service worker it should be inside the codebase, otherwise you can put the sw in the public folder and use inject manifest strategy (in that case the sw must use importScripts having the injection point or disable it in the pwa options)

**Reply by userquin** [reply]
https://vite-pwa-org.netlify.app/guide/cookbook.html

**Reply by orangsed** [reply]
okay, thank u so much sir
